packages/server-wallet/src/wallet/wallet.ts

```diff
@@ -239,8 +239,7 @@ export class Wallet extends EventEmitter<WalletEvents> {
   public async jumpStartObjectives(): Promise<ObjectiveResult[]> {
     const objectives = await this._engine.getApprovedObjectives();
     const objectiveIds = objectives.map(o => o.objectiveId);
-    // Instead of getting messages per objective we just get them all at once
-    // This will prevent us from querying the database for each objective
+    await this._engine.store.updateObjectiveProgressDate(objectiveIds);
```

Comment:
This could be a single query? ie. `this._engine.store.updateObjectiveProgressData()` can
- filter by the approved objectives
- update the progress data
- return the data that `syncObjectives` returns

Reply:
coming back to this, I don't think it's worth making this a single query unless it's easy.